File: leetcode/Algorithms/Easy/1 Two Sum/Solution.c
#include <stdlib.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) 
{
    *returnSize = 2;
    int *outArr = malloc(*returnSize * sizeof(int));

    for (int i = 0; i < numsSize - 1; i++)
    {
        for (int j = i + 1; j < numsSize; j++)
        {
            if (nums[i] + nums[j] == target)
            {
                outArr[0] = i;
                outArr[1] = j;

                return outArr;
            }
            
        }
        
    }

    return outArr;
}

typedef struct Entry
{
    int key;
    int value;
    struct Entry* next;
} Entry;


typedef struct Hashmap
{
    Entry** table;
    int size;
} Hashmap;

File: leetcode/Algorithms/Easy/1 Two Sum/solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        sum_dict = {}

        for i in range(len(nums)):
            needed_num = target - nums[i]
            if sum_dict.get(needed_num) is not None:
                ans.append(sum_dict.get(needed_num))
                ans.append(i)
                return ans
            
            sum_dict[nums[i]] = i
        
        return ans

    

nums = [2,7,11,15]
target = 9

s = Solution()
print(s.twoSum(nums, target))